### PR TITLE
[ceph_mds] Collect mds specific data in ceph_mds

### PR DESCRIPTION
**Pull request by nkshirsagar**
Enhance the ceph_mds plugin to collect data specific
to mds nodes.

Related: #1945
Resolves: #2800

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [x] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?

### CODE REVIEW

File: sos/report/plugins/ceph_mds.py [hunk 1]
```diff
@@ -45,4 +45,60 @@ def setup(self):
...
+        # If containerized, run commands in containers
+        try:
+            cname = self.get_all_containers_by_regex("ceph-mds*")[0][1]
+        except Exception:
+            cname = None
+
+        self.add_cmd_output([
+            "ceph daemon %s %s"
+            % (mdsid, cmd) for mdsid in mds_ids for cmd in ceph_cmds
+        ], container=cname)
```
Comment (TurboTurtle) [inline sos/report/plugins/ceph_mds.py:101]: This looks fine by itself, however how many mds ids would a "normal" environment potentially have here?

If there's a decent number of them, we may want to put these collections in an id-specific subdir within the plugin directory, for better organization.

**Review by TurboTurtle — COMMENTED** [review]
Overall this looks fine, just a couple minor comments.

File: sos/report/plugins/ceph_mds.py [hunk 2]
```diff
@@ -25,7 +25,7 @@ def setup(self):
         })
 
         self.add_copy_spec([
-            "/var/log/ceph.log",
+            # "/var/log/ceph.log",
```
[the remark this answers is not in the document]
Comment (nkshirsagar) [reply sos/report/plugins/ceph_mds.py:28]: ack

File: sos/report/plugins/ceph_mds.py (hunk 1 above)
Comment (nkshirsagar) [reply sos/report/plugins/ceph_mds.py:101]: I would suppose there would not be more than one, in most cases. I will check with @hillpd to confirm.

**Comment by taodd** [comment]
@nkshirsagar Nikhil, 

There are some cephfs admin socket commands that can not be collected 
E.g dump_cache
Many fs production cluster has millions or 10's millions of inode cached in the memory
The impact would be:
1. The dump will take a very long time, file output file will be very large
2. could slow down the mds significantly 

There are also some command require an extra parameter from user
like dump_inode and dirfrag ls ..

Most options looks good to me
My suggestion is to remove below:
dump_cache
dump_inode
dirfrag ls
dump snaps
log dump
openfiles ls [this doesn't work as expected any more, would be misleading, though I added :)]

**Comment by nkshirsagar** [comment]
> 

I've made the changes you suggested. Thanks!

**Review by taodd — APPROVED** [review]

File: sos/report/plugins/ceph_mds.py (hunk 1 above)
Comment (nkshirsagar) [reply sos/report/plugins/ceph_mds.py:94]: @hillpd would there ever be multiple mds ids on a single node?

Comment (hillpd) [reply sos/report/plugins/ceph_mds.py:94]: Normal environments will not have more than one MDS on a single node.

**Review by TurboTurtle — APPROVED** [review]

**Review by pmoravec — APPROVED** [review]

**Pull request merged**